core/include/usFramework.h

```diff
@@ -28,74 +28,176 @@
+     *    Any exceptions that occur during bundle starting are wrapped in a
+     *    \c std::runtime_error and then published as a framework event of type
+     *    {@link FrameworkEvent#ERROR}
+     * -# This Framework's state is set to {@link #STATE_ACTIVE}.
+     * -# A framework event of type {@link FrameworkEvent#STARTED} is fired
      *
      * @throws std::runtime_error If this Framework could not be started.
      */
     void Start();
 
     /**
+     * Start this Framework.
+     *
+     * <p>
+     * Calling this method is the same as calling {@link #Start()}. There are no
+     * start options for the Framework.
+     *
+     * @param options Ignored. There are no start options for the Framework.
+     * @throws std::runtime_error If this Framework could not be started.
+     * @see #Start()
+     */
+    void Start(uint32_t options);
+
+    /**
      * Stop this Framework.
      *
-     * The following steps are taken to stop this Framework:
-     * -# This Framework's state is set to STOPPING.
-     * -# All installed bundles must be stopped.
+     * <p>
+     * The method returns immediately to the caller after initiating the
+     * following steps to be taken on another thread. If the Framework was not
+     * build with threading enabled, the steps are executed in the main thread.
+     * -# This Framework's state is set to {@link #STATE_STOPPING}.
+     * -# All installed bundles are stopped without changing each bundle's
+     *    persistent <i>autostart setting</i>. Any exceptions that occur
+     *    during bundle stopping are wrapped in a \c std::runtime_error and
+     *    then published as a framework event of type {@link FrameworkEvent#ERROR}
      * -# Unregister all services registered by this Framework.
      * -# Event handling is disabled.
-     * -# All resources held by this Framework are released. This includes threads, open files, etc.
+     * -# This Framework's state is set to {@link #STATE_RESOLVED}.
+     * -# All resources held by this Framework are released. This includes
+     *    threads, open files, etc.
+     * -# Notify all threads that are waiting at {@link #WaitForStop(const std::chrono::milliseconds&)
+     *    WaitForStop} that the stop operation has completed.
      *
-     * After being stopped, this Framework may be discarded or started.
+     * <p>
+     * After being stopped, this Framework may be discarded, initialized or
+     * started.
      *
      * @throws std::runtime_error If stopping this Framework could not be initiated.
      */
     void Stop();
 
     /**
+     * Stop this Framework.
+     *
+     * <p>
+     * Calling this method is the same as calling {@link #Stop()}. There are no
+     * stop options for the Framework.
+     *
+     * @param options Ignored. There are no stop options for the Framework.
+     * @throws std::runtime_error If stopping this Framework could not be
+     *         initiated.
+     * @see #Stop()
+     */
+    void Stop(uint32_t options);
```

Comment:
Simply for the sake of conciseness and maintenance, ```void Stop(uint32_t options);``` could be modified to take a default value of ```0``` and ```void Stop();``` could then be eliminated completely. This could also be applied to the base class ```Bundle```.